src/go/wsl-helper/pkg/wsl-utils/version_windows.go

```diff
@@ -211,87 +210,83 @@ func getPackageVersion(fullName string) (*PackageVersion, error) {
 	return nil, fmt.Errorf("no info found for %s", fullName)
 }
 
-// runWSLExe runs WSL.exe and returns the standard output.
-// This can be replaced for testing.
-func runWSLExe(ctx context.Context, args ...string) (string, error) {
-	systemDir, err := windows.GetSystemDirectory()
-	if err != nil {
-		return "", fmt.Errorf("failed to get system directory: %w", err)
-	}
-	wslPath := filepath.Join(systemDir, "wsl.exe")
-	cmd := exec.CommandContext(ctx, wslPath, "--status")
-	cmd.Stderr = os.Stderr
-	stdout, err := cmd.Output()
-	if err != nil {
-		return "", fmt.Errorf("failed to get WSL status: %w", err)
-	}
-	output := windows.UTF16PtrToString(
-		(*uint16)(unsafe.Pointer(unsafe.SliceData(append(stdout, 0, 0)))),
-	)
-	return output, nil
-}
-
 // isInboxWSLInstalled checks if the "in-box" version of WSL is installed,
 // returning whether it's installed, and whether the kernel is installed
-func isInboxWSLInstalled(ctx context.Context) (bool, bool, error) {
-	runWSLExeFunc := runWSLExe
+func isInboxWSLInstalled(ctx context.Context, log *logrus.Entry) (bool, bool, error) {
+	var allErrors []error
+
+	// Check if the core is installed
+	coreInstalled := false
+	newRunnerFunc := NewWSLRunner
 	if f := ctx.Value(&kWSLExeOverride); f != nil {
```

Comment:
I missed the PR from a month ago that introduced these context keys, but could you explain how exactly `kWSLExeOverride` and `kUpgradeCodeOverride` are modified? Because I see them being queried with `ctx.Value(&kWSLExeOverride)` -- does that update them based on the context?

Reply:
They're only set in the test:
https://github.com/rancher-sandbox/rancher-desktop/blob/d9b322461dcd07400f99451e58c38876698dd8b0/src/go/wsl-helper/pkg/wsl-utils/version_windows_test.go#L82
https://github.com/rancher-sandbox/rancher-desktop/blob/d9b322461dcd07400f99451e58c38876698dd8b0/src/go/wsl-helper/pkg/wsl-utils/version_windows_test.go#L101

I didn't want to actually run commands in the test, because I didn't want to depend on the host system's WSL installation state during testing (and modify it…)